data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessor.java

```diff
@@ -0,0 +1,296 @@
+        this.numberOfRecordsFailedCounter = pluginMetrics.counter(NUMBER_OF_RECORDS_FLUSHED_TO_LAMBDA_FAILED);
+        this.lambdaLatencyMetric = pluginMetrics.timer(LAMBDA_LATENCY_METRIC);
+        this.requestPayload = pluginMetrics.gauge(REQUEST_PAYLOAD_SIZE, new AtomicLong());
+        this.responsePayload = pluginMetrics.gauge(RESPONSE_PAYLOAD_SIZE, new AtomicLong());
+
+        functionName = lambdaProcessorConfig.getFunctionName();
+        whenCondition = lambdaProcessorConfig.getWhenCondition();
+        maxRetries = lambdaProcessorConfig.getMaxConnectionRetries();
+        batchOptions = lambdaProcessorConfig.getBatchOptions();
+        payloadModel = lambdaProcessorConfig.getPayloadModel();
+        codecContext = new OutputCodecContext();
+
+        if (payloadModel.equals(BATCH_EVENT)) {
+            JsonOutputCodecConfig jsonOutputCodecConfig = new JsonOutputCodecConfig();
+            jsonOutputCodecConfig.setKeyName(batchOptions.getKeyName());
+            codec = new JsonOutputCodec(jsonOutputCodecConfig);
+            maxEvents = batchOptions.getThresholdOptions().getEventCount();
+            maxBytes = batchOptions.getThresholdOptions().getMaximumSize(); // remove
+            maxCollectionDuration = batchOptions.getThresholdOptions().getEventCollectTimeOut();
+            isBatchEnabled = true;
+            LOG.info("maxEvents:" + maxEvents + " maxbytes:" + maxBytes + " maxDuration:" + maxCollectionDuration);
+        } else if(payloadModel.equals(SINGLE_EVENT)) {
+            NdjsonOutputConfig ndjsonOutputCodecConfig = new NdjsonOutputConfig();
+            codec = new NdjsonOutputCodec(ndjsonOutputCodecConfig);
+            isBatchEnabled = false;
+        } else{
+            throw new RuntimeException("invalid payload_model option");
+        }
+
+        if(!lambdaProcessorConfig.getInvocationType().equals(LambdaCommonConfig.EVENT) &&
+                !lambdaProcessorConfig.getInvocationType().equals(LambdaCommonConfig.REQUEST_RESPONSE)){
+            throw new RuntimeException("Unsupported invocation type " + lambdaProcessorConfig.getInvocationType());
+        }
+
+        //Translate dataprepper invocation type to lambda invocation type
+        Map<String, String> invocationTypeMap = Map.of(
+                LambdaCommonConfig.EVENT, EVENT_LAMBDA,
+                LambdaCommonConfig.REQUEST_RESPONSE, REQUEST_RESPONSE_LAMBDA
+        );
+        invocationType = invocationTypeMap.get(lambdaProcessorConfig.getInvocationType());
+
+        bufferedEventHandles = new LinkedList<>();
+        events = new ArrayList();
+
+        lambdaClient = LambdaClientFactory.createLambdaClient(lambdaProcessorConfig.getAwsAuthenticationOptions(),
+                lambdaProcessorConfig.getMaxConnectionRetries()
+                , awsCredentialsSupplier);
+
+        this.bufferFactory = new InMemoryBufferFactory();
+        try {
+            currentBuffer = this.bufferFactory.getBuffer(lambdaClient, functionName, invocationType);
```

Comment:
The buffer is overloaded in DataPrepper. Looks like this is not just buffer but tightly coupled with lambda. We should consider renaming this class and interface to be clear.

Reply:
i am handling the same way we would do it in the sink. i can address refactor in another pr.